rootston/xwayland.c

```diff
@@ -174,29 +173,24 @@ static void handle_request_maximize(struct wl_listener *listener, void *data) {
 static void handle_map_notify(struct wl_listener *listener, void *data) {
 	struct roots_xwayland_surface *roots_surface =
 		wl_container_of(listener, roots_surface, map_notify);
+	struct wlr_xwayland_surface *xsurface = data;
 	struct roots_view *view = roots_surface->view;
-	struct wlr_xwayland_surface *xsurface = view->xwayland_surface;
-	struct roots_desktop *desktop = view->desktop;
+	//struct roots_desktop *desktop = view->desktop;
 
 	view->wlr_surface = xsurface->surface;
 	view->x = (double)xsurface->x;
 	view->y = (double)xsurface->y;
 
-	wlr_list_push(desktop->views, roots_surface->view);
```

Comment:
I don't really understand what to do here, the view has already been added here: https://github.com/swaywm/wlroots/pull/427/files#diff-94b621ef8193066933ec606392245a3bL253

Reply:
I think the wayland surface is destroyed on unmap and a new wayland surface is created on map.